ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::stream_coordinator::IndexerStreamCoordinator;
+use aptos_api::context::Context;
+use aptos_config::config::NodeConfig;
+use aptos_logger::{error, info};
+use aptos_mempool::MempoolClientSender;
+use aptos_moving_average::MovingAverage;
+use aptos_protos::datastream::v1::{
+    indexer_stream_server::{IndexerStream, IndexerStreamServer},
+    raw_datastream_response,
+    stream_status::StatusType,
+    RawDatastreamRequest, RawDatastreamResponse, StreamStatus,
+};
+use aptos_storage_interface::DbReader;
+use aptos_types::chain_id::ChainId;
+use futures::Stream;
+use std::{net::ToSocketAddrs, pin::Pin, sync::Arc};
+use tokio::{runtime::Runtime, sync::mpsc};
+use tokio_stream::wrappers::ReceiverStream;
+use tonic::{transport::Server, Request, Response, Status};
+
+// Default Values
+pub const DEFAULT_NUM_RETRIES: usize = 3;
+pub const RETRY_TIME_MILLIS: u64 = 300;
+const TRANSACTION_CHANNEL_SIZE: usize = 35;
+const DEFAULT_EMIT_SIZE: usize = 1000;
+
+type ResponseStream = Pin<Box<dyn Stream<Item = Result<RawDatastreamResponse, Status>> + Send>>;
+
+// The GRPC server
+pub struct IndexerStreamService {
+    pub context: Arc<Context>,
+    pub processor_task_count: u16,
+    pub processor_batch_size: u16,
+    pub output_batch_size: u16,
+}
+
+/// Creates a runtime which creates a thread pool which sets up the grpc streaming service
+/// Returns corresponding Tokio runtime
+pub fn bootstrap(
+    config: &NodeConfig,
+    chain_id: ChainId,
+    db: Arc<dyn DbReader>,
+    mp_sender: MempoolClientSender,
+) -> Option<Runtime> {
+    if !config.indexer_grpc.enabled {
+        return None;
+    }
```

Comment:
This feels kinda weird... like why did we even get here if this isn't enabled? :shrug: 
wouldn't mind seeing an assert here...

Reply:
The existing way that the node is structured requires this b/c we just call all the possible runtimes in a single file (think API, logging, indexer, etc.) and only does something if the config enables it. We can move this check upstream I guess but that's not the existing pattern.